### ISSUE TITLE
[JENKINS-24904] Add option to archive artifacts in the background

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#5862]
<p>I have a job that takes about 2 hours to run — 15 minutes to build, 1 hour to run automated tests, and 45 minutes to archive the artifacts (the node's upstream internet connection isn't very fast).</p>

<p>Would it make sense to add the option to start archiving the artifacts right after a certain build step completes, and archive in parallel with the remaining build steps?  In a case like this, it could reduce the job from 2 hours to 1 hour 15 minutes.</p>

---
<details><summary><i>Originally reported by <a title="Steve Mokris" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=smokris">smokris</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-24904" target="_blank">Add option to archive artifacts in the background</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2017-03-09T04:43:58+00:00
<li><b>imported</b>: 2025-11-05
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>I don't think this would make sense. Too error prone.</p>
